docs/design/simplified_federation_deployment.md

```diff
@@ -0,0 +1,388 @@
+  federation controller manager and it also uses the same official
+  `hyperkube` release image as the federation API server.
+* Kubeconfig required by the federation controller manager to discover
+  and authenticate with the federation API server is referenced as a
+  secret in the pod template spec and mounted as a volume inside the
+  container.
+* The mounted kubeconfig file is passed as an argument to the federation
+  controller manager's `--kubeconfig` flag.
+
+##### Additional enhancements
+
+Users can pass `--register` as an additional flag to `kubeadm init
+federation` to also register the cluster with federation. See the
+[Cluster Registration/Deregistration](#cluster-
+registrationderegistration) section below for details on how this is
+done.
+
+##### Caveats
+
+* In the first phase of this implementation, we only create a single
+  instance of `etcd`. This neither makes `etcd` nor the federation
+  control plane HA. HA federation control plane is on the road map and
+  will arrive in one of the future releases.
+* Federation control plane components aren't started until additional
+  nodes join the bootstrap cluster. In other words, these components
+  aren't pinned to the bootstrap cluster's master node. We just create
+  their manifests at this stage and they are started when additional
+  schedulable nodes join the bootstrap cluster.
+
+
+#### Cluster Registration/Deregistration
+
+`kubectl` is Kubernetes' client-side CLI used for ongoing day-to-day
+interactions with a set of clusters. `kubectl` is already being extended
+to enable interactions with federation. Users can also already use
+`kubectl` to register clusters with and deregister clusters from
+federation. As already described, this process is manual, tedious and
+error-prone. Hence, we aim to reduce these processes to a single command
+each.
+
+##### Registration
+
+When a user runs `kubectl register <cluster-context-name> --bootstrap-
+cluster=<bootstrap-cluster-context>`, we perform the following operations
+assuming that the current kubeconfig context is a federation endpoint:
+
+**1. Create a kubeconfig secret for the cluster in federation**
+
+* Read the kubeconfig fields: `cluster`, `user` and `context`
+  corresponding to the context `<cluster-context-name>` into a separate
+  kubeconfig and create a secret for that in the the bootstrap cluster
+  indicated by the `--bootstrap-cluster` flag.
+
+**2. Retrieve the cluster endpoint**
+
+* Read the API server endpoint for `<cluster-context-name>` from the
+  `cluster` field in the local kubeconfig file.
+
+**3. Create the cluster resource in federation**
+
+* Using the `<cluster-context-name>`'s API server endpoint and the
+  kubeconfig secret name, create the cluster API resource in federation.
+
+##### Deregistration
+
+When a user runs `kubectl unregister <cluster-context-name> --bootstrap-
+cluster`, we perform the following operations, again assuming that the
```

Comment:
I thought that was implicit from the previous section, but updated it for clarity here. Thanks for that feedback!